### PR TITLE
op-node,contracts-bedrock: setup `loadAllocs` usage

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Modularize the `op-node` command for creating L2 genesis
blocks such that it can read a starting L1 block from disk
instead of needing to fetch it from a node. This makes the
process more simple and reproducible, because the block
JSON file can be written to disk and committed into a repo.

Also add the script for calling the `op-node` L2 genesis generation
to `contracts-bedrock`. It will create the L2 genesis state
in `contracts-bedrock/.testdata` which will be read by
`vm.readAllocs(string)` to populate the initial state.

### CODE REVIEW

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go:87]: Nice. I like this pattern of encapsulating all the flags in a var.

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go:142]: Also happy to see this move to `Path` over `String`.

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go:186]: Was just talking to Seb about how we handle errors, I think this is a good style change.

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go]: is there a particular reason we use `ctx.String("l1-rpc")` here over using the `l1RPC` var?

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go]: Speaking of error messages, is it worth adding a little to these `fmt.Errorf`s or do you think they'll be self explanatory in the context through which they bubble up?

File: op-node/cmd/genesis/cmd.go
Comment (EvanJRichard) [inline op-node/cmd/genesis/cmd.go:274]: To make sure I follow, we define our own serialization structs here rather than importing them because the RPC could really be coming from anywhere, right?

**Review by EvanJRichard — APPROVED** [review]
Smells good to me. Left a couple comments and questions. I'll also add the same disclaimer that I'm putting on my first handful of reviews: consider waiting for review from someone with more background context.

File: op-node/cmd/genesis/cmd.go
Comment (tynes) [reply op-node/cmd/genesis/cmd.go]: great point we should use that variable, this is left over from a previous iteration

File: op-node/cmd/genesis/cmd.go
Comment (tynes) [reply op-node/cmd/genesis/cmd.go]: these def need more context

File: op-node/cmd/genesis/cmd.go
Comment (tynes) [reply op-node/cmd/genesis/cmd.go:274]: geth defines them in an `internal` package so we cannot import them unfortunately

**Review by Inphi — APPROVED** [review]
LGTM!

**Comment by tynes** [comment]
Thank you for the reviews!

**Pull request merged**
